### DISCUSSION TITLE
addition of Advanced tablet driver settings

### DISCUSSION THREAD

**Discussion by TheFantasticLoki** [ppy/osu#19751] [Ideas]
On mobile, sorry. There have been a few things that have always bugged me since the addition of OTD being built into lazer. The major inconsistency between built in OTD and standalone OTD, whether this be the inability to add filters or the fact the play area display does not match the actual tablet dimensions or the fact that the area ratio is completely different. Overall there are just a lot of issues with the built in driver that makes me outright disable it on install.

I feel there are MANY ways of addressing this issue but I'll only be listing 1 way of going about things as typing on mobile is very aggravating.
1. The first thing that I feel MUST BE CHANGED is the area ratio display. OTD standalone already has a perfect, easy to understand ratio display that most of the community already knows how to read and utilize, this works by simply showing you the width to height ratio by dividing the width by the height and on most monitors this is read out as 1.778. This is half implemented already as the aspect ratio slider shows the proper value but the area preview does not and shows 7:4 on 1.78 ratio which is confusing. Peppy... please fix
2. No way to add or configure filters in lazer OTD. Filters are an essential utility in OTD standalone of which is mostly used to add smoothness to the tablet output to remove jitter but can and is used to accomplish other results such as circular area and even adding chatter to some devices. While this is not used by the majority of the community it is still an essential feature as it allows users with extremely jittery tablets or even Parkinson's disease to have a playable experience. Please add this feature back into the built in OTD Driver.
3. Going off purely memory from this point as not home to verify and no tablet settings on mobile to check off of. Compounding on problem 1 is the lock aspect ratio feature is not properly implemented IMO. This is a feature of OTD that is not used enough imo as it's not extremely well known but you can lock your area ratio to the **ratio of the display you are using** preventing non uniform tablet areas. What I mean is that many players accidentally use an area with a ratio that does not match their screen meaning that moving 1 unit on the area will move more or less than 1 unit on screen. This is usually done when a player just gets a tablet for the first time or when a newer player hits a skill wall. In my experience this usually takes the form of a square area on a rectangle monitor in order to more easily hit cross screen jumps or just notes on the edges of the play area in general. This causes inconsistent cursor movement which causes issues on more complex maps meaning players will encounter much harder skill walls far more often without them even realizing why. I personally have fixed over 40 players areas just by locking the ratio to the monitor and in every case except for 1 the player immediately saw improvement and much more consistency in their aim AND tapping ability as they were not having to focus nearly as hard on aiming. 
The way in which this is implemented in lazer is that it locks your aspect ratio to the currently set ratio which could be completely wrong to begin with. This is further compounded as an issue with the way the ratio is displayed on the area preview AND the fact that your monitor/game display ratio is not displayed anywhere in game for players to match.

This is as much as my sanity can take writing up on mobile but I will try to add more later when I can make it home and verify other problems still exist but my main 3 gripes are listed so I think this will do for now. Also for those curious the only case in which locking the players area ratio did not work was an instance in which the player had learned to play with both a square area AND his area rotated 90 degrees. Just couldn't cope with such an extreme change.

QUICK EDIT: If you are not going to change the area ratio display there should ABSOLUTELY be a display of what your monitors ratio is so players at least have a chance of matching it up properly.

EDIT 2: Was able to double check some things on lazer since I am home and I adjusted the post to contain only currently existing issues.

**Comment by TheFantasticLoki** [comment]
I seen that there is a conform to current game ratio button BUT it's not setting the proper ratio. With my existing standalone otd width and height which is 1:1 ratio with my monitor, In Game displays my ratio on the AREA PREVIEW as 7:4 but when I conform to game ratio it sets it to 9:5 which is incorrect. I also double checked to make sure I had no screen scaling of any kind.

**Comment by peppy** [comment]
Please either:

- Search first
- Use less capitals
- Only state the basics of an issue

or

- Stop posting

Most of what you say, even after editing, is duplicating existing discussions.
